modules/manage/pages/manage-xdcr/replicate-using-scopes-and-collections.adoc

```diff
@@ -0,0 +1,178 @@
++
+image::manage-xdcr/xdcr-basic-replication-definition-to-102.png[,560,align=left]
++
+Note the confirmatory notification that appears underneath the replication-definition.
+As this indicates, if a replication is defined to include any destination-entity &#8212; bucket, scope, or collection &#8212; that does not exist, the entity will be ignored, and no attempt will be made to replicate data to it.
+However, if other specified entities are valid, replication to them will proceed.
+
+. To migrate data, switch on the *Migrate collections* toggle, in the middle of the screen:
++
+image::manage-xdcr/xdcr-migrate-collections-toggle.png[,520,align=left]
++
+Three new fields thus appear, which allow migration to be defined.
+*Replication Filter for Source* allows a _regular expression_ to be specified, whereby only a subset of documents within `travel-sample` are replicated.
+*Replicate to Collection* allows specification of a collection on the target cluster: the collection must be preceded by the name of the scope that contains it, with scope-name and collection-name comma-separated.
+The *Save Mapping* button allows the migration-definition to be saved.
+
+. Specify that _airline_ documents from `travel-sample` be replicated to the collection `airlineScope.airlineCollection`.
+Use the regular expression `REGEXP_CONTAINS(META().id, "^airline")`.
+The fields now appear as follows:
++
+image::manage-xdcr/xdcr-migrate-collections-definition.png[,520,align=left]
++
+Left-click on the *Save Mapping* button, to save the mapping:
++
+image::manage-xdcr/xdcr-save-mapping-button.png[,150,align=left]
++
+Note that the saved rule now appears in the *Mapping Rules* column, at the upper right of the screen:
++
+image::manage-xdcr/mapping-rules-migration-definition.png[,200,align=left]
+
+. Save the replication, by left-clicking on the *Save Replication* button, at the bottom of the screen:
++
+image::manage-xdcr/saveReplicationButton.png[,140,align=Left]
++
+The *XDCR Replications* screen now returns, with the *Outgoing Replications* panel appearing as follows:
++
+image::manage-xdcr/xdcr-outgoing-replication-migration.png[,680,align=left]
++
+As this indicates, the defined replication is now proceeding from `travel-sample` on the source cluster, to `ts` on the remote.
+
+. On cluster `10.144.210.102`, access the *Buckets* screen, and left-click on the *Documents* tab for the collection `airlineCollection`, within the bucket `ts`.
++
+image::manage-xdcr/xdcr-access-usAirlineCollection.png[,680,align=left]
++
+The documents within the collection are now displayed:
++
+image::manage-xdcr/xdcr-target-collection-filled.png[,680,align=left]
++
+This indicates that the `airline` documents from `travel-sample` have been successfully filtered and replicated to the `airlineCollection` collection, within the remote bucket `ts`.
+
+[#replicate-data-between-collections]
+== Replicate Data Between Collections
+
+XDCR allows data to be replicated between collections.
```

Comment:
In this section, it looks like it forces the user to *have* to specify collections mappings to be able to replicate collections (explicit mapping), which shouldn't be the case. XDCR was designed so that implicit mapping would be offered first. Implicit mapping should be intuitive enough and hides a lot of the management complexity from the user. Only if the implicit mapping is not enough, should the user opt in for explicit mapping.

I would recommend a section to showcase implicit mapping and try to pursuade the users to go that route. Then, explicit mapping can come in for those who need a more powerful mapping experience.

I have sent you a copy of the blog draft, which explains the concepts.

One point to note for implicit mapping: XDCR monitors the target bucket for created/removed collections at a rate of once per minute. If a target collection was missing when a replication is set up, and then created, it may take up to a minute for XDCR to replicate data to the target collection.

Reply:
I'll put in implicit examples first, and try to explain the concepts a bit more. (I'll leave the major explanation for the work I do based on your blog.)